apps/lib/apps.c

```diff
@@ -829,7 +832,18 @@ static const char *format2string(int format)
 }
 
 /* Set type expectation, but clear it if objects of different types expected. */
-#define SET_EXPECT(expect, val) ((expect) = (expect) < 0 ? (val) : ((expect) == (val) ? (val) : 0))
+#define SET_EXPECT(val) \
+    (expect = expect < 0 ? (val) : (expect == (val) ? (val) : 0))
+#define SET_EXPECT1(pvar, val) \
+    if ((pvar) != NULL) { \
```

Comment:
For an internal macro like this, we often dispense with the parenthesis around the arguments.  Not a significant point and not a request to change anything.